### PR TITLE
Add SD/MMC support for FRDM-K64F

### PR DESCRIPTION
**Pull request by tilk**
This pull request allows to build and run the `fat_fs` example on the Kinetis FRDM-K64F board. The DTS files are modified to include the SPI1 port and its connection to the microSD card slot on the board.

### CODE REVIEW

**Review by danieldegrasse — COMMENTED** [review]
A few comments:
- You need to clean up your commit message, see [Zephyr's commit guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines)
- You will likely need to exclude the `frdm_k64f` from the fatfs filesystem test, see the `platform_exclude` directive [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/fs/fat_fs/sample.yaml#L17) ([documentation on this directive](https://docs.zephyrproject.org/3.0.0/guides/test/twister.html))

Other than that, this contribution looks good to me

**Comment by tilk** [comment]
Added missing `Signed-off-by`, hope it's the last thing needed to merge the PR.

**Comment by danieldegrasse** [comment]
@tilk If you are on a linux machine, running the `check_compliance.py` script may help you eliminate some of these issues. You can run it like this: `python3 ${ZEPHYR_BASE}/scripts/ci/check_compliance.py -c main..HEAD` to check all commits between your copy of `main` and your current `HEAD`, or run it without `-c` to only check the latest commit.

**Comment by danieldegrasse** [comment]
@tilk I believe you can resolve this issue by running `west update`.

**Comment by tilk** [comment]
@danieldegrasse This indeed solved the issue, thank you.

**Review by danieldegrasse — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

File: boards/arm/frdm_k64f/frdm_k64f.dts [hunk 1]
```diff
@@ -146,6 +146,26 @@ arduino_spi: &spi0 {
...
+	status = "okay";
+	pinctrl-0 = <&spi1_default>;
+	pinctrl-names = "default";
+	cs-gpios = <&gpioe 4 GPIO_ACTIVE_LOW>;
+
+	sdhc0: sdhc@0 {
+		compatible = "zephyr,sdhc-spi-slot";
+		reg = <0>;
+		status = "okay";
+		label = "SDHC_0";
```
Comment (stephanosio) [inline boards/arm/frdm_k64f/frdm_k64f.dts:159]: [This broke main](https://github.com/zephyrproject-rtos/zephyr/runs/8239856646?check_suite_focus=true#step:11:22157):
> devicetree error: 'label' is marked as deprecated in 'properties:' in /__w/zephyr/zephyr/dts/bindings/sdhc/zephyr,sdhc-spi-slot.yaml for node /soc/spi@4002d000/sdhc@0.

File: boards/arm/frdm_k64f/frdm_k64f.dts [hunk 2]
```diff
@@ -146,6 +146,26 @@ arduino_spi: &spi0 {
 	pinctrl-names = "default";
 };
 
+&spi1 {
+	status = "okay";
+	pinctrl-0 = <&spi1_default>;
+	pinctrl-names = "default";
+	cs-gpios = <&gpioe 4 GPIO_ACTIVE_LOW>;
+
+	sdhc0: sdhc@0 {
```
Comment (stephanosio) [inline boards/arm/frdm_k64f/frdm_k64f.dts:155]: [another breakage](https://github.com/zephyrproject-rtos/zephyr/runs/8244898222?check_suite_focus=true#step:12:291):

> devicetree error: Label 'sdhc0' appears on /soc/spi@4002c000/sdhc@1 and on /soc/spi@4002d000/sdhc@0

**Pull request merged**